src/workers/commit-extractor.ts

```diff
@@ -30,3 +30,37 @@ export class GithubPullRequestExtractor implements CommitExtractor {
     return stream;
   }
 }
+
+export class GithubShaExtractor implements CommitExtractor {
+  octokit: Octokit;
+  owner: string;
+
+  constructor(octokit: Octokit, owner: string) {
+    this.octokit = octokit;
+    this.owner = owner;
+  }
+
+  commits(sha: string, repository: string): Observable<string[]> {
+    const promise = this.octokit.repos.getCommit({
+      owner: this.owner,
+      repo: repository,
+      ref: sha
+    });
+
+    const stream = from(promise)
+      .pipe(
+        flatMap((x) =>
+          this.octokit.repos.listCommits({
+            owner: this.owner,
+            repo: repository,
+            // eslint-disable-next-line @typescript-eslint/camelcase
+            since: x.data.commit.author.date
+          })
+        )
```

Comment:
Note, as we can't `listCommits` from a certain SHA until now (but only the other way around, commits that are older than an SHA), I used a trick here in which I use the SHA to first fetch a certain commit, and then only will I list the commits **_since_** said commit (this includes the commit itself that we fetched by hand)